### PR TITLE
Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

### PR DESCRIPTION
**Pull request by xichengliudui**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref #https://github.com/kubernetes/kubernetes/pull/76121

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xichengliudui** [comment]
/cc @BenTheElder 
This pr has been open for a long time and can be merged directly.

**Comment by xichengliudui** [comment]
a new pr was opened to resolve conflicts based on the original.

File: hack/jenkins/upload-to-gcs.sh [hunk 1]
```diff
@@ -121,15 +121,16 @@ function find_version() {
     echo "${KUBE_GIT_VERSION-}"
   else
     # Last resort from the started.json
-    gsutil cat ${gcs_build_path}/started.json 2>/dev/null |\
+    gsutil cat "${gcs_build_path}"/started.json 2>/dev/null |\
```
Comment (BenTheElder) [inline hack/jenkins/upload-to-gcs.sh]: nit: 
```suggestion
    gsutil cat "${gcs_build_path}/started.json" 2>/dev/null |\
```

Comment (xichengliudui) [reply hack/jenkins/upload-to-gcs.sh]: Has been updated.

Comment (xichengliudui) [reply hack/jenkins/upload-to-gcs.sh]: Has been updated.

File: hack/jenkins/test-dockerized.sh [hunk 2]
```diff
@@ -21,7 +21,12 @@ set -o xtrace
 
 retry() {
   for i in {1..5}; do
-    "$@" && return 0 || sleep "${i}"
+    if "$@"
```
Comment (fejta) [inline hack/jenkins/test-dockerized.sh]: typically we put if and then on the same line: `if "$@"; then`

**Review by fejta — COMMENTED** [review]
/lgtm
/approve

Thanks!

File: hack/jenkins/test-dockerized.sh (hunk 2 above)
Comment (xichengliudui) [reply hack/jenkins/test-dockerized.sh]: YES, has been updated.

**Review by fejta — COMMENTED** [review]
/lgtm

**Pull request merged**